### PR TITLE
Fixed issue #18655: Data loss if field in response table it missing

### PR DESCRIPTION
**Pull request by gabrieljenik**
New method: setAllAttributes.

### CODE REVIEW

**Comment by Shnoulle** [comment]
this mean setAttributes have more attribute than setAllAttributes ?

Maybe name is not good ? Or fix can be different ?

The previous issue was about setAttributes with token attributes inside ?  setTableAttributes for the new name ?

**Comment by gabrieljenik** [comment]
> this mean setAttributes have more attribute than setAllAttributes ?

setAllAttributes is a brand new method.
We can name it differently if prefer.

> The previous issue was about setAttributes with token attributes inside ? setTableAttributes for the new name ?

I prefer to fix that on another ticket. The original issue is a blocking one that needs to be solved.

**Review by Shnoulle — APPROVED** [review]
Unsure on the name , but don't found better idea :)

:+1:

File: application/models/Response.php [hunk 1]
```diff
@@ -255,4 +255,24 @@ public static function getEncryptedAttributes($surveyid = 0)
         }
         return $aAttributes;
     }
+
+    /**
+     * Set all the specified attributes.
+     * If any attribute doesn't exist in the DB, an Exception is thrown.
+     * @inhe
```
Comment (olleharstedt) [inline application/models/Response.php:262]: inhe? :)

**Review by olleharstedt — CHANGES_REQUESTED** [review]

**Comment by olleharstedt** [comment]
> @olleharstedt Does this requires changes still?

No idea, ping Carsten on Cliq?

**Comment by gabrieljenik** [comment]
> > @olleharstedt Does this requires changes still?
> 
> No idea, ping Carsten on Cliq?

Ohh, I mention it as you were the one requesting changes.
I will ping Carsten still. Thanks!

**Pull request merged**